crates/bevy_math/src/common_traits.rs

```diff
@@ -245,10 +320,55 @@ impl NormedVectorSpace for f32 {
     fn norm(self) -> f32 {
         ops::abs(self)
     }
+}
 
+impl NormedVectorSpace for DVec4 {
     #[inline]
-    fn norm_squared(self) -> f32 {
-        self * self
+    fn norm(self) -> f64 {
+        self.length()
+    }
+
+    #[inline]
+    fn norm_squared(self) -> f64 {
+        self.length_squared()
+    }
+}
+
+impl NormedVectorSpace for DVec3 {
+    #[inline]
+    fn norm(self) -> f64 {
+        self.length()
+    }
+
+    #[inline]
+    fn norm_squared(self) -> f64 {
+        self.length_squared()
+    }
+}
+
+impl NormedVectorSpace for DVec2 {
+    #[inline]
+    fn norm(self) -> f64 {
+        self.length()
+    }
+
+    #[inline]
+    fn norm_squared(self) -> f64 {
+        self.length_squared()
+    }
+}
+
+impl NormedVectorSpace for f64 {
+    #[inline]
+    #[cfg(feature = "std")]
+    fn norm(self) -> f64 {
+        f64::abs(self)
+    }
+
+    #[inline]
+    #[cfg(all(any(feature = "libm", feature = "nostd-libm"), not(feature = "std")))]
```

Comment:
Why are the conditional cfg's outside the function?
Could they annotate just the line inside?

Reply:
I tried that, but I ran into compile errors even with only `std` or `libm` enabled. Not entirely sure why, it seems like it would be valid. Quite possible the problem is between my ears.